backend/fms_core/tests/test_template_importers/library_preparation_test.py

```diff
@@ -0,0 +1,220 @@
+            strandedness='Double stranded',
+            library_comment='Library 1 Comment'
+        )
+
+        self.library_2 = dict(
+            **self.library_batch_1,
+            library_container_barcode='Container4Library2',
+            library_container_name='Container4Library2',
+            library_container_kind='Tube',
+            volume_used=4,
+            library_volume=3,
+            index=self.index_2,
+            strandedness='Single stranded',
+            library_comment='Library 2 Comment'
+        )
+
+        self.library_3 = dict(
+            **self.library_batch_2,
+            library_container_barcode='Container4Library3',
+            library_container_name='Container4Library3',
+            library_container_kind='Tube',
+            volume_used=5,
+            library_volume=5,
+            index=self.index_3,
+            strandedness='Double stranded',
+            library_comment='Library 3 Comment'
+        )
+
+        self.prefill_data()
+
+
+    def prefill_data(self):
+        sample_kind, _ = SampleKind.objects.get_or_create(name='DNA')
+
+        (container, errors, warnings) = create_container(barcode=self.source_sample_container_barcode,
+                                                         kind='Tube',
+                                                         name='Container4LibraryPrep')
+
+        create_full_sample(name=self.source_sample_name, volume=self.source_sample_initial_volume, concentration=25,
+                           collection_site='TestCaseSite', creation_date=datetime.datetime(2022, 1, 15, 0, 0),
+                           container=container, sample_kind=sample_kind)
+
+
+    def test_import(self):
+        # Basic test for all templates - checks that template is valid
+        result = load_template(importer=self.importer, file=self.file)
+        self.assertEqual(result['valid'], True)
+
+        # Test source sample
+        source_sample, _, _ = get_sample_from_container(barcode=self.source_sample_container_barcode)
+        new_volume = self.source_sample_initial_volume - self.library_1['volume_used'] - self.library_2['volume_used'] - self.library_3['volume_used']
+        self.assertEqual(source_sample.volume, new_volume)
+
+        # Test first library
+        sample_library_1, _, _ = get_sample_from_container(barcode='Container4Library1')
+        library_1 = sample_library_1.derived_sample_not_pool.library
+
+        self.assertEqual(sample_library_1.volume, self.library_1['library_volume'])
+
+        # Library info tests
+        self.assertEqual(library_1.library_type, self.library_1['library_type'])
+        self.assertEqual(library_1.platform, self.library_1['platform'])
+        self.assertEqual(library_1.index, self.library_1['index'])
+        self.assertEqual(library_1.strandedness, self.library_1['strandedness'])
+
+        # Process and process measurements tests
+        pm_1 = ProcessMeasurement.objects.get(source_sample=source_sample,
```

Comment:
Test somewhere in here that the process for library_1 and library_2 is the same...